version: "3.9"
networks:
  devops:
    driver: bridge

volumes:
  postgres_data:
services:
  db:
    image: postgres
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_NAME=${POSTGRES_NAME}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    healthcheck:
      test: /usr/bin/pg_isready -u postgres
      interval: 5s
      timeout: 10s
      retries: 30
    networks:
      - devops
    profiles:
      - backend
      - app
  web:
    build:
      context: ./django-app
    volumes:
      - ./django-app:/code
    environment:
      - POSTGRES_NAME=${POSTGRES_NAME}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - DATABASE_HOST=db
    depends_on:
      - db
    networks:
      - devops
    profiles:
      - frontend
      - app

  proxy:
    build:
      context: ./docker-nginx
    volumes:
      - ./docker-nginx/default.conf:/etc/nginx/conf.d/default.conf:rw
    ports:
      - 80:80
    depends_on:
      - web
      - postgres
    networks:
      - devops
    profiles:
      - frontend
      - app



